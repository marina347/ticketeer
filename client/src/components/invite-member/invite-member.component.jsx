import React from "react";
import { connect } from "react-redux";

import FormButton from "../form-button/form-button.component";
import { generateHashedBoardIdAsync } from "../../redux/board/board.actions";
import { selectBoardLink } from "../../redux/board/board.selectors";
import { selectToken } from "../../redux/user/user.selectors";
import { InviteContainer } from "./invite-member.styles";
import EnvVariables from "../../env-variables";

export const InviteMember = ({ boardLink, generateLink, boardId, token }) => {
  const boardUrl = `${EnvVariables.REACT_APP_SERVER_PATH}/home/boards/join-board/${boardLink}`;

  return (
    <InviteContainer>
      <p>Create link for joining this board!</p>
      <div id="board-link">{boardLink !== "" ? boardUrl : boardLink}</div>
      <FormButton
        id="board-link-button"
        onClick={() => generateLink(boardId, token)}
      >
        CREATE
      </FormButton>
    </InviteContainer>
  );
};

const mapStateToProps = (state, ownProps) => ({
  boardLink: selectBoardLink(ownProps.boardId)(state),
  token: selectToken(state),
});

const mapDispatchToProps = (dispatch) => ({
  generateLink: (boardId, token) =>
    dispatch(generateHashedBoardIdAsync(boardId, token)),
});

export default connect(mapStateToProps, mapDispatchToProps)(InviteMember);
