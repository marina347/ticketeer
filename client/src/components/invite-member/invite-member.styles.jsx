import styled from "styled-components";

export const InviteContainer = styled.div`
  margin: 15px;
  margin-left: 30px;
`;

InviteContainer.displayName = "InviteContainer";
