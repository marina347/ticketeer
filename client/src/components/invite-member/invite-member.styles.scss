.icon-box {
  cursor: pointer;
}
.icon {
  fill: var(--color-white);
  width: 3.8rem;
  height: 3.8rem;
}
