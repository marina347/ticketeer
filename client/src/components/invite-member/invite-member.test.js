import React from "react";
import { shallow } from "enzyme";
import { InviteMember } from "./invite-member.component";
import EnvVariables from "../../env-variables";

describe("InviteMember component", () => {
  let wrapper;

  const mockGenerateLink = jest.fn();
  const boardLink = "";

  beforeEach(() => {
    const mockProps = {
      generateLink: mockGenerateLink,
      boardLink,
    };

    wrapper = shallow(<InviteMember {...mockProps} />);
  });

  it("Should render InviteMember component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("Should render empty board link in div", () => {
    const boardLink = wrapper
      .find("InviteContainer")
      .find('[id="board-link"]')
      .text();
    expect(boardLink).toBe("");
  });

  it("Should render board link in div", () => {
    const newMockProps = {
      generateLink: mockGenerateLink,
      boardLink: "1234",
    };

    const newWrapper = shallow(<InviteMember {...newMockProps} />);
    const boardLink = newWrapper
      .find("InviteContainer")
      .find('[id="board-link"]')
      .text();
    expect(boardLink).toBe(
      `${EnvVariables.REACT_APP_SERVER_PATH}/home/boards/join-board/1234`
    );
  });

  it("Form button should have prop onClick", () => {
    expect(
      wrapper
        .find("InviteContainer")
        .find('[id="board-link-button"]')
        .prop("onClick")
    ).not.toEqual(undefined);
  });
});
