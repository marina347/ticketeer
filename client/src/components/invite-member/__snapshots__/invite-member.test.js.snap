// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InviteMember component Should render InviteMember component 1`] = `
<InviteContainer>
  <p>
    Create link for joining this board!
  </p>
  <div
    id="board-link"
  />
  <FormButton
    id="board-link-button"
    onClick={[Function]}
  >
    CREATE
  </FormButton>
</InviteContainer>
`;
